plugins/registration.ts

```diff
@@ -17,53 +18,115 @@ interface RegistrationFeatures {
     /** The username used to register the client to the server. */
     username?: string;
 
+    /** The password for the user account associated with the username field. */
+    password?: string;
+
     /** The realname used to register the client to the server. */
     realname?: string;
 
-    /** The password used to connect the client to the server. */
-    password?: string;
+    /** Optional server password. */
+    serverPassword?: string;
+
+    /** The authentication method to use. Defaults to NickServ if omitted.
+     * * `NickServ` - Non-standard nickserv authentication.
+     * * `sasl` - SASL PLAIN auth. Errors out if SASL fails.
+     * * `saslThenNickServ` - Try SASL PLAIN, but fallback to NickServ if it fails.
+     */
+    authMethod?: "NickServ" | "sasl" | "saslThenNickServ";
   };
   state: {
     user: User;
   };
 }
 
+function* chunks(str: string, n: number): Generator<string, void> {
+  for (let i = 0; i < str.length; i += n) {
+    yield str.slice(i, i + n);
+  }
+}
```

Comment:
What is the goal of using a generator here, instead of just returning the array containing the chunks?

Reply:
I got the snippet out of an SO answer and wasn't sure if the chunks() function might get used again somewhere else where it might be better to get the chunks as they come. No other reason, just a bit of (embarrassingly so) premature optimization. I'll get rid of it.